src/App.tsx

```diff
@@ -15,36 +18,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Default,
+  Name,
+  Length,
+}
+
+type SortTheGoods = (
+  items: Good[],
+  sortBy: SortBy | null,
+  option: boolean,
+) => Good[];
+
+const getSortedGoods: SortTheGoods = (goods, sortBy, isReverseAdded) => {
+  const sortedGoods: Good[] = [...goods];
+
+  if (sortBy) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortBy.Name:
+          return good1.localeCompare(good2);
+
+        case SortBy.Length:
+          return good1.length - good2.length;
+
+        default:
+          return SortBy.Default;
+      }
+    });
+  }
+
+  if (isReverseAdded) {
+    return sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortBy | null>(null);
```

Comment:
instead of making null default you could use SortBy.Default
```suggestion
  const [sortBy, setSortBy] = useState<SortBy>(SortBy.Default);
```

Reply:
> instead of making null default you could use SortBy.Default

Good point